src/microbe_stage/editor/CellEditorComponent.cs

```diff
@@ -1037,9 +1037,33 @@ public float CalculateHitpoints()
         return maxHitpoints;
     }
 
-    public float CalculateStorage()
+    public float GetNominalCapacity()
     {
-        return MicrobeInternalCalculations.CalculateCapacity(editedMicrobeOrganelles);
+        return editedMicrobeOrganelles.Sum(o => MicrobeInternalCalculations
+            .GetNominalCapacityForOrganelle(o.Upgrades, o.Definition));
+    }
+
+    public Dictionary<Compound, float> GetAdditionalCapacities()
+    {
+        var dict = new Dictionary<Compound, float>();
+
+        foreach (var organelle in editedMicrobeOrganelles)
+        {
+            var capacity = MicrobeInternalCalculations
+                .GetAdditionalCapacityForOrganelle(organelle.Upgrades, organelle.Definition);
+
+            if (capacity.Compound == null)
+                continue;
+
+            if (dict.TryGetValue(capacity.Compound, out var currentCapacity))
+                dict[capacity.Compound] = currentCapacity + capacity.Capacity;
+            else
+                dict.Add(capacity.Compound, capacity.Capacity);
+        }
+
+        var nominalCap = GetNominalCapacity();
+        return dict.Select(e => new KeyValuePair<Compound, float>(e.Key, e.Value + nominalCap))
```

Comment:
Instead of needing to redo the dictionary you could get the nominal capacity at the start of this method and then in the line 1061 do `dict.Add(capacity.Compound, capacity.Capacity + nominalCap)`

Reply:
Problem with that approach is that if two organelles have specific capacity for the same compound, the nominal capacity will be added twice if I understand the suggestion correctly

Reply:
The TryGetValue returns true *only* when the value exists, so the `Add` call is made only on the first time (when TryGetValue returns false). But feel free to point out if I misunderstood.

Reply:
Yeah, you're right, my bad